### PR TITLE
Add support for untyped prometheus metrics

### PR DESCRIPTION
**Pull request by mvollman**
Add support to the prometheus metricbeat module for reporting untyped
metrics from prometheus exporters.

### CODE REVIEW

**Comment by jsoriano** [comment]
Thanks @mvollman for this contribution! It LGTM, could you please add a changelog entry?

@exekias could you take a look to this PR too?

**Comment by jsoriano** [comment]
jenkins, test this please

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -88,6 +88,8 @@ https://github.com/elastic/beats/compare/v6.4.0...master[Check the HEAD diff]
 - Avoid mapping issues in kubernetes module. {pull}8487[8487]
 - Fix issue that would prevent kafka module to find a proper broker when port is not set {pull}8613[8613]
 - Fix range colors in multiple visualizations. {issue}8633[8633] {pull}8634[8634]
+- Add Kafka dashboard. {pull}8457[8457]
```
Comment (jsoriano) [inline CHANGELOG.asciidoc]: Not related to this change

File: metricbeat/module/prometheus/collector/collector_test.go [hunk 2]
```diff
@@ -150,6 +150,34 @@ func TestGetPromEventsFromMetricFamily(t *testing.T) {
...
+				Type: dto.MetricType_UNTYPED.Enum(),
+				Metric: []*dto.Metric{
+					{
+						Label: []*dto.LabelPair{
+							{
+								Name:  proto.String("handler"),
+								Value: proto.String("query"),
+							},
+						},
+						Counter: &dto.Untyped{
```
Comment (exekias) [inline metricbeat/module/prometheus/collector/collector_test.go]: This should be:

```suggestion
						Untyped: &dto.Untyped{
```

File: metricbeat/module/prometheus/collector/collector_test.go [hunk 3]
```diff
@@ -150,6 +150,34 @@ func TestGetPromEventsFromMetricFamily(t *testing.T) {
...
+							Value: proto.Float64(10),
+						},
+					},
+				},
+			},
+			Event: PromEvent{
+				key: "http_request_duration_microseconds",
+				value: common.MapStr{
+					"value": int64(10),
```
Comment (exekias) [inline metricbeat/module/prometheus/collector/collector_test.go]: and:
```suggestion
					"value": float64(10),
```

**Comment by jsoriano** [comment]
jenkins, test this please

**Comment by mvollman** [comment]
@jsoriano I didn't see what caused beats-ci to fail and so I am not sure what to do to fix the issue.  Can I just duplicate your jenkins comment to get a retest so I can see the error message?

**Comment by jsoriano** [comment]
@mvollman we can ignore the error in jenkins CI build, it seems caused by #8412.

**Comment by mvollman** [comment]
ok great thanks.  @exekias when you get a chance can you take another look at the changes?

**Comment by exekias** [comment]
Thank you @mvollman :tada: :tada:  This will go out with Metricbeat 6.6

**Pull request merged**